src/test/cypress/support/requests/CourseManagementRequests.ts

```diff
@@ -53,22 +53,33 @@ export class CourseManagementRequests {
 
     /**
      * Creates a course with the specified title and short name.
-     * @param course the response object from a previous call to createCourse
      * @param title the title of the programming exercise
      * @param programmingShortName the short name of the programming exercise
      * @param packageName the package name of the programming exercise
      * @param releaseDate when the programming exercise should be available (default is now)
      * @param dueDate when the programming exercise should be due (default is now + 1 day)
+     * @param body an object containing either the course or exercise group the exercise will be added to
      * @returns <Chainable> request response
      */
-    createProgrammingExercise(course: any, title: string, programmingShortName: string, packageName: string, releaseDate = new Date(), dueDate = new Date(Date.now() + oneDay)) {
-        const programmingTemplate = programmingExerciseTemplate;
+    createProgrammingExercise(
+        title: string,
+        programmingShortName: string,
+        packageName: string,
+        body: { course: any } | { exerciseGroup: any },
+        releaseDate = new Date(),
+        dueDate = new Date(Date.now() + oneDay),
+    ) {
+        const isExamExercise = body.hasOwnProperty('exerciseGroup');
+        const programmingTemplate: any = this.getCourseOrExamExercise(programmingExerciseTemplate, body);
         programmingTemplate.title = title;
         programmingTemplate.shortName = programmingShortName;
         programmingTemplate.packageName = packageName;
-        programmingTemplate.releaseDate = releaseDate.toISOString();
-        programmingTemplate.dueDate = dueDate.toISOString();
-        programmingTemplate.course = course;
+        if (isExamExercise) {
+            programmingTemplate.releaseDate = releaseDate.toISOString();
+            programmingTemplate.dueDate = dueDate.toISOString();
+        } else {
+            programmingTemplate.allowComplaintsForAutomaticAssessments = true;
+        }
```

Comment:
This looks like it is the wrong way around. Is that correct?

Reply:
This also looked wrong to me, so I checked with a [pre-merge commit](https://github.com/ls1intum/Artemis/blob/c3382b33725c464cf4e337cc14947494145540ca/src/test/cypress/support/requests/CourseManagementRequests.ts) and the condition is the same there. @rs-maker please verify whether or not this is correct. I also think that the condition should be inverted

Reply:
I inverted the condition. Setting release- and due date for exam exercises makes no sense. I ran the programming exercise related tests locally against ts2 and they still work.
In this branch there are no tests which actually create a programming exercise for exams. Furthermore it is valid to create a programming exercise without those dates, which is why we didn't get any errors in the tests.